catkin_ws/src/camera_driver/src/camera_node_sequence.py

```diff
@@ -102,19 +150,23 @@ def grabAndPublish(self, stream, publisher):
             image_msg.header.frame_id = self.frame_id
             publisher.publish(image_msg)
 
+            self.camera_info_msg.header.stamp = stamp
+            cam_info_publisher.publish(self.camera_info_msg)
             # Clear stream
             stream.seek(0)
             stream.truncate()
 
             if not self.has_published:
-                rospy.loginfo("[%s] Published the first image." % (self.node_name))
+                rospy.loginfo("[%s] Published the first image." %
+                              (self.node_name))
                 self.has_published = True
 
-            rospy.sleep(rospy.Duration.from_sec(0.001))
+            rospy.sleep(rospy.Duration.from_sec(0.005))
```

Comment:
also here no need for a 1000 loops per seconds. 200 is enough to get a smooth frame rate (never bigger than 30 so nyquist rests happy)

Reply:
@AmaurX I don't think this is looping, it's a generator that is being called by `self.camera.capture_sequence` at the set framerate.